### PR TITLE
Add shared examples for check compliance action.

### PR DESCRIPTION
**Pull request by lfu**
To dry up the specs.

@miq-bot assign @abellotti

### CODE REVIEW

File: spec/requests/vms_spec.rb [hunk 1]
```diff
@@ -1416,93 +1416,7 @@ def update_raw_power_state(state, *vms)
...
-            "href"    => api_vm_url(nil, vm2)
-          )
-        )
-      }
-
-      expect(response.parsed_body).to include(expected)
-      expect(response).to have_http_status(:ok)
-    end
-  end
+  it_behaves_like "has check compliance action", "vm", :vm_vmware, "Vm"
```
Comment (abellotti) [inline spec/requests/vms_spec.rb:1419]: can you rename this, sounds a bit weird in the rspec -fd output, maybe rename both to it_behaves_like "_it implements a check compliance action_"

Comment (lfu) [reply spec/requests/vms_spec.rb]: Now it looks like:
```
  behaves like a check compliance action
    to a single vm without compliance policies assigned
    to an invalid vm without appropriate role
    to multiple vms without appropriate role
    to a single vm with a compliance policy
    to multiple vms
    to an invalid vm
```

Comment (abellotti) [reply spec/requests/vms_spec.rb]: 👍 Thanks @lfu for the update and the refactoring, should make it easier for adding it to the other endpoints.

**Comment by simaishi** [comment]
Jansa backport details:

```
$ git log -1
commit 27bd8f78724051f9dc8096fc77818d73e1d7ac6b
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Thu Apr 16 17:14:11 2020 -0400

    Merge pull request #802 from lfu/check_compliance_shared_examples

    Add shared examples for check compliance action.

    (cherry picked from commit c491cffaa778df5e70914bbe23be0df4e82f6648)
```

**Pull request merged**